src/node/request.ts

```diff
@@ -0,0 +1,74 @@
+const https = require('https');
+const http = require('http');
+
+import type { ClientRequest, RequestOptions, ServerResponse } from 'http';
+import { isDev } from '../env';
+
+const protocol = isDev ? http : https;
+
+const options = {
+  hostname: import.meta.env.VITE_API_BASE_URL,
+  port: isDev ? 8080 : undefined,
+  headers: { 'Content-Type': 'application/json' },
+};
+
+function request(options: RequestOptions) {
+  return new Promise((resolve, reject) => {
+    const req: ClientRequest = protocol.request(
+      options,
+      (res: ServerResponse) => {
+        let data = '';
+
+        // Collect the response data
+        res.on('data', (chunk) => {
+          data += chunk;
+        });
+
+        // Resolve or Reject the Promise when the response ends
+        res.on('end', () => {
+          const { statusCode } = res;
+          if (statusCode >= 200 && statusCode < 300) {
+            try {
+              const response = JSON.parse(data);
+              resolve(response);
+            } catch (error) {
+              reject({
+                statusCode: 500,
+                message: `Failed to parse response: ${(error as Error).message}`,
+              });
+            }
+          } else {
+            const error = {
+              statusCode,
+              message: data,
```

Comment:
why are we also not parsing the data here, should also be a JSON in case of an error?

imo:

1. parse data in try
2. depending on the status code reject or approve
3. if parsing fails do same as now